src/d3d9/d3d9_device.cpp

```diff
@@ -4187,7 +4183,7 @@ namespace dxvk {
 
     DxvkBufferSliceHandle physSlice;
 
-    if (Flags & D3DLOCK_DISCARD) {
+    if ((Flags & D3DLOCK_DISCARD) && needsReadback) {
```

Comment:
Yup, `needsReadback` basically tracks whether the resource has been written by the GPU. If that's not the case, we don't need to discard because we don't have to synchronize anyway. So we might as well save some memory.